src/Smidge.Core/Models/BundleExtensions.cs

```diff
@@ -11,21 +12,42 @@ public static class BundleExtensions
         /// <param name="bundleMgr"></param>
         /// <param name="debug"></param>
         /// <returns></returns>
+        [Obsolete("Use GetBundleOptions(IBundleManager, string) and specify a configuration profile name.")]
         public static BundleOptions GetBundleOptions(this Bundle bundle, IBundleManager bundleMgr, bool debug)
         {
             var bundleOptions = debug
-                ? (bundle.BundleOptions == null ? bundleMgr.DefaultBundleOptions.DebugOptions : bundle.BundleOptions.DebugOptions)
-                : (bundle.BundleOptions == null ? bundleMgr.DefaultBundleOptions.ProductionOptions : bundle.BundleOptions.ProductionOptions);
+                ? GetBundleOptions(bundle, bundleMgr, SmidgeOptionsProfile.Debug)
+                : GetBundleOptions(bundle, bundleMgr, SmidgeOptionsProfile.Default);
 
             return bundleOptions;
         }
+        
+        /// <summary>
+        /// Get the bundle options from the bundle if they have been set otherwise with the defaults
+        /// </summary>
+        /// <param name="bundle"></param>
+        /// <param name="bundleMgr"></param>
+        /// <param name="profileName"></param>
+        /// <returns></returns>
+        public static BundleOptions GetBundleOptions(this Bundle bundle, IBundleManager bundleMgr, string profileName)
+        {
+            var bundleOptions = bundle.BundleOptions == null
+                ? bundleMgr.DefaultBundleOptions[profileName]
```

Comment:
Do we want to check the profileName exists here with TryGetValue(profileName, out var options) instead of assuming the key exists?

Reply:
This doesn't assume the profile exists. The BundleEnvironmentOptions collection guarantees that a BundleOptions instance will be returned when asking for a BundleOptions instance via the indexer property (a new instance will be created and stored in the collection if a profile with the requested name doesn't yet exist). (See BundleEnvironmentOptions indexer)

In the case you really want to know if a named profile exists (and without automatically creating one if it doesnt) you can call BundleEnvironmentOptions.TryGetProfileOptions();